#! /bin/sh
patch -p1 -l -f $* < $0
exit $?


diff -Nrub gpgme-1.10.0.orig/lang/cpp/src/data.h gpgme-1.10.0/lang/cpp/src/data.h
--- gpgme-1.10.0.orig/lang/cpp/src/data.h       2018-05-02 09:27:30.675672414 +0200
+++ gpgme-1.10.0/lang/cpp/src/data.h    2018-05-02 09:34:35.790237498 +0200
@@ -31,6 +31,10 @@
 #include <algorithm>
 #include <memory>

+#ifndef ssize_t
+#define ssize_t int
+#endif
+
 namespace GpgME
 {

diff -Nrub gpgme-1.10.0.orig/lang/cpp/src/interfaces/dataprovider.h gpgme-1.10.0/lang/cpp/src/interfaces/dataprovider.h
--- gpgme-1.10.0.orig/lang/cpp/src/interfaces/dataprovider.h    2018-05-02 09:27:30.674672434 +0200
+++ gpgme-1.10.0/lang/cpp/src/interfaces/dataprovider.h 2018-05-02 09:35:00.879739684 +0200
@@ -31,6 +31,10 @@

 #include <gpg-error.h>

+#ifndef ssize_t
+#define ssize_t int
+#endif
+
 namespace GpgME
 {

diff -Nrub gpgme-1.10.0.orig/lang/cpp/src/key.h gpgme-1.10.0/lang/cpp/src/key.h
--- gpgme-1.10.0.orig/lang/cpp/src/key.h        2018-05-02 09:27:30.675672414 +0200
+++ gpgme-1.10.0/lang/cpp/src/key.h     2018-05-02 09:35:39.617971059 +0200
@@ -30,7 +30,7 @@
 #include "gpgmefw.h"

 #include <memory>
-#include <sys/time.h>
+#include <time.h>

 #include <vector>
 #include <algorithm>
diff -Nrub gpgme-1.10.0.orig/lang/qt/src/dataprovider.h gpgme-1.10.0/lang/qt/src/dataprovider.h
--- gpgme-1.10.0.orig/lang/qt/src/dataprovider.h        2018-05-02 09:27:30.672672474 +0200
+++ gpgme-1.10.0/lang/qt/src/dataprovider.h     2018-05-02 09:33:53.348079615 +0200
@@ -30,6 +30,9 @@

 #include <QtCore/QByteArray>

+#ifndef ssize_t
+#define ssize_t int
+#endif

 class QIODevice;

diff -Nrub gpgme-1.10.0.orig/lang/qt/src/encryptjob.h gpgme-1.10.0/lang/qt/src/encryptjob.h
--- gpgme-1.10.0.orig/lang/qt/src/encryptjob.h  2018-05-02 09:27:30.672672474 +0200
+++ gpgme-1.10.0/lang/qt/src/encryptjob.h       2018-05-02 09:33:14.761845224 +0200
@@ -90,7 +90,7 @@
        not be performed, but full validity assumed for all keys
        without further checks.
     */
-    virtual QGPGME_DEPRECATED_EXPORT GpgME::Error start(const std::vector<GpgME::Key> &recipients,
+    virtual GpgME::Error start(const std::vector<GpgME::Key> &recipients,
             const QByteArray &plainText, bool alwaysTrust = false) = 0;

     /*!
diff -Nrub gpgme-1.10.0.orig/lang/qt/src/signjob.h gpgme-1.10.0/lang/qt/src/signjob.h
--- gpgme-1.10.0.orig/lang/qt/src/signjob.h     2018-05-02 09:27:30.672672474 +0200
+++ gpgme-1.10.0/lang/qt/src/signjob.h  2018-05-02 09:33:35.037442925 +0200
@@ -85,7 +85,7 @@
        Starts the signing operation. \a signers is the list of keys to
        sign \a plainText with. Empty (null) keys are ignored.
     */
-    virtual QGPGME_DEPRECATED_EXPORT GpgME::Error start(const std::vector<GpgME::Key> &signers,
+    virtual GpgME::Error start(const std::vector<GpgME::Key> &signers,
             const QByteArray &plainText,
             GpgME::SignatureMode mode) = 0;
